### PR TITLE
refactor: initial refactor of code

### PR DESCRIPTION
**Pull request by H4ad**
TODOs:

- [x] Finish the tests
- [x] Cleanup package.json
- [x] Add tsup

### CODE REVIEW

**Comment by H4ad** [comment]
If you want to collaborate on this, we can merge this one and continue in the next PRs.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
+# fd-slicer
```
Comment (kibertoad) [inline README.md]:
```suggestion
# better-fd-slicer
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+#### Events
+
+##### 'error'
+
+Emitted if `fs.close` returns an error when auto closing.
+
+##### 'close'
+
+Emitted when fd-slicer closes the file descriptor due to `autoClose`. Never
+emitted if created from a buffer.
```
Comment (kibertoad) [inline README.md:199]: can you add `## Credits` section, with the reference to the original project?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
+{
+  "name": "fd-slicer",
```
Comment (kibertoad) [inline package.json]:
```suggestion
  "name": "better-fd-slicer",
```

File: package.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  "version": "1.1.0",
+  "description": "safely create multiple ReadStream or WriteStream objects from the same file descriptor",
+  "main": "index.js",
+  "scripts": {
+    "test": "mocha --reporter spec --check-leaks",
+    "test-cov": "istanbul cover node_modules/mocha/bin/_mocha -- --reporter dot --check-leaks test/index.spec.ts",
+    "test-travis": "istanbul cover node_modules/mocha/bin/_mocha --report lcovonly -- --timeout 10000 --reporter spec --check-leaks test/index.spec.ts"
+  },
+  "author": "Andrew Kelley <superjoe30@gmail.com>",
```
Comment (kibertoad) [inline package.json:30]: can you replace this with your own credentials?

**Comment by kibertoad** [comment]
@H4ad let's iterate on bigger things in follow-up PRs, but would be great to address attribution in this one already

**Comment by H4ad** [comment]
I will only have time at night but I will address those comments, thanks for the review!

**Comment by kibertoad** [comment]
@H4ad any updates on this? I went to Prague for the weekend, but will resume my work on the project on Tuesday.

**Comment by H4ad** [comment]
I will finish the details tonight, we can release on Monday.

File: package.json (hunk 4 above)
Comment (H4ad) [reply package.json:30]: I added the maintainer's field instead of changing the author.

I think it worth to keep the original author.

**Comment by H4ad** [comment]
Some notes:

- I will rewrite: https://github.com/andrewrk/node-pend to remove this dependency.
- I will rewrite/remove https://www.npmjs.com/package/streamsink

You can merge (not squash) as it is, but I will do those missing rewrites just to keep the entire code typed and simpler (without deps).

File: package.json (hunk 4 above)
Comment (kibertoad) [reply package.json:30]: makes sense!

**Review by kibertoad — APPROVED** [review]

**Comment by kibertoad** [comment]
sounds like a plan!

**Comment by kibertoad** [comment]
@H4ad can you add GitHub Actions before modifying any logic?

**Comment by H4ad** [comment]
Sure! We also can release the first version of this library, we don't need to wait my refactoring to release.

**Pull request merged**